client/checkout/blocks/upe-fields.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * External dependencies
+ */
+import {
+	Elements,
+	ElementsConsumer,
+	PaymentElement,
+} from '@stripe/react-stripe-js';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import confirmUPEPayment from './confirm-upe-payment.js';
+import { getConfig } from 'utils/checkout';
+import { PAYMENT_METHOD_NAME_CARD } from '../constants.js';
+
+const WCPayUPEFields = ( {
+	api,
+	activePaymentMethod,
+	stripe,
+	elements,
+	billing: { billingData },
+	eventRegistration: {
+		onPaymentProcessing,
+		onCheckoutAfterProcessingWithSuccess,
+	},
+	emitResponse,
+	shouldSavePayment,
+} ) => {
+	const [ paymentIntentId, setPaymentIntentId ] = useState( null );
+	const [ clientSecret, setClientSecret ] = useState( null );
+	const [ hasRequestedIntent, setHasRequestedIntent ] = useState( false );
+	const [ isUPEComplete, setIsUPEComplete ] = useState( false );
+	const [ selectedUPEPaymentType, setSelectedUPEPaymentType ] = useState(
+		''
+	);
+
+	const businessName = getConfig( 'accountDescriptor' );
+	const paymentMethodsConfig = getConfig( 'paymentMethodsConfig' );
+	const testMode = getConfig( 'testMode' );
+	const testCopy = (
+		<p>
+			<strong>Test mode:</strong> use the test VISA card 4242424242424242
+			with any expiry date and CVC.
+		</p>
+	);
+
+	useEffect( () => {
+		if ( paymentIntentId || hasRequestedIntent ) {
+			return;
+		}
+
+		async function createIntent() {
+			const response = await api.createIntent();
```

Comment:
There's an uncaught exception here when `api.createIntent` throws an error. This error just gets dropped in the console and there's nothing displayed in the UI, which is a little confusing (this happened to me, because I had the wrong currency selected--which I know is being fixed in another issue). Maybe do you fancy wrapping this in a `try/catch` or something and returning the error to the UI?

Reply:
I've got a try/catch on this now. However, displaying the error message proved a little more difficult than I expected as it seems you can only "bubble up" errors during the defined events, which would require the customer to try to submit the form first. I found a workaround that isn't super pretty, but at least gets the message across clearly.